File: rock63/src/main/java/com/uksusoff/rock63/jobs/RockJobCreator.java
package com.uksusoff.rock63.jobs;

import com.evernote.android.job.Job;
import com.evernote.android.job.JobCreator;

public class RockJobCreator implements JobCreator {

    @Override
    public Job create(String tag) {
        switch (tag) {
            case NotificationJob.TAG:
                return new NotificationJob();
            case DataUpdateJob.TAG:
                return new DataUpdateJob();
            default:
                return null;
        }
    }

}

File: rock63/src/main/java/com/uksusoff/rock63/jobs/RockJobCreator.kt
package com.uksusoff.rock63.jobs

import com.evernote.android.job.Job
import com.evernote.android.job.JobCreator

/**
 * Created by User on 28.08.2016.
 */
class RockJobCreator : JobCreator {
    override fun create(tag: String): Job? {
        return when (tag) {
            NotificationJob.TAG -> NotificationJob()
            DataUpdateJob.TAG -> DataUpdateJob()
            else -> null
        }
    }
}